### PR TITLE
Docker more info 

### PR DESCRIPTION
**Pull request by radames**

### CODE REVIEW

File: docs/hub/spaces-sdks-docker.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Docker Spaces
 
-Spaces accommodate custom [Docker containers](https://docs.docker.com/get-started/) for apps that go outside the scope of Streamlit and Gradio. Docker Spaces give freedom to users to go beyond the limits of what was previously possible with the standard sdks. From FastAPI and Go endpoints to Phoenix apps and ML Ops tools, Docker Spaces can help in many different setups.
+Spaces accommodate custom [Docker containers](https://docs.docker.com/get-started/) for apps outside the scope of Streamlit and Gradio. Docker Spaces allows users to go beyond the limits of what was previously possible with the standard sdks. From FastAPI and Go endpoints to Phoenix apps and ML Ops tools, Docker Spaces can help in many different setups.
```
Comment (julien-c) [inline docs/hub/spaces-sdks-docker.md:3]: ~~maybe let's revert this to keep this PR focused on the main changes?~~ no big deal

File: docs/hub/spaces-sdks-docker.md [hunk 2]
```diff
@@ -17,6 +17,7 @@ app_port: 7860
 ---
 ```
 
+Internally you could have as many open ports as you want - and use a reverse proxy like Nginx to dispatch requests from the broader internet.
```
Comment (julien-c) [inline docs/hub/spaces-sdks-docker.md:20]:
```suggestion
Internally you could have as many open ports as you want, for instance you can install Elasticsearch inside your Space and call it internally on its default port 9200.

If you want to expose apps served on multiple ports to the outside world, a workaround is to use a reverse proxy like Nginx to dispatch requests from the broader internet (on a single port) to different internal ports.
```

File: docs/hub/spaces-sdks-docker.md [hunk 3]
```diff
@@ -57,7 +58,11 @@ WORKDIR $HOME/app
 COPY --chown=user . $HOME/app
 ```
 
+## Data Persistence
+
+We don't provide a way to persist data in the Docker Spaces. However, we suggest using our [Datasets hub](datasets) for specific cases, where you can store state and data in a git LFS repository. In all other cases, you might need an external storage solution DB, S3, etc.
```
Comment (julien-c) [inline docs/hub/spaces-sdks-docker.md:63]:
```suggestion
We don't yet provide a built-in way to persist data in the Docker Spaces. However, we suggest using our [Datasets hub](datasets) for specific cases, where you can store state and data in a git LFS repository. In other cases, you might want to use from your Space's code an external storage solution like an external hosted DB, S3, etc.
```

**Review by julien-c — COMMENTED** [review]
left some suggestions

File: docs/hub/spaces-sdks-docker.md (hunk 1 above)
Comment (osanseviero) [inline docs/hub/spaces-sdks-docker.md:3]:
```suggestion
Spaces accommodate custom [Docker containers](https://docs.docker.com/get-started/) for apps outside the scope of Streamlit and Gradio. Docker Spaces allow users to go beyond the limits of what was previously possible with the standard SDKs. From FastAPI and Go endpoints to Phoenix apps and ML Ops tools, Docker Spaces can help in many different setups.
```

File: docs/hub/spaces-sdks-docker.md [hunk 4]
```diff
@@ -17,6 +17,9 @@ app_port: 7860
 ---
 ```
 
+Internally you could have as many open ports as you want, for instance you can install Elasticsearch inside your Space and call it internally on its default port 9200.
```
Comment (osanseviero) [inline docs/hub/spaces-sdks-docker.md:20]:
```suggestion
Internally you could have as many open ports as you want. For instance, you can install Elasticsearch inside your Space and call it internally on its default port 9200.
```

File: docs/hub/spaces-sdks-docker.md [hunk 5]
```diff
@@ -57,7 +60,11 @@ WORKDIR $HOME/app
 COPY --chown=user . $HOME/app
 ```
 
+## Data Persistence
+
+We don't yet provide a built-in way to persist data in the Docker Spaces. However, we suggest using our [Datasets hub](datasets) for specific cases, where you can store state and data in a git LFS repository. In other cases, you might want to use from your Space's code an external storage solution like an external hosted DB, S3, etc.
```
Comment (osanseviero) [inline docs/hub/spaces-sdks-docker.md:65]:
```suggestion
We don't yet provide a built-in way to persist data in the Docker Spaces. However, we suggest using our Datasets Hub for specific cases, where you can store state and data in a git LFS repository. You can find an example of persistence [here](https://huggingface.co/spaces/julien-c/persistent-data), which uses the [`huggingface_hub` library](https://huggingface.co/docs/huggingface_hub/index) for programmatically uploading files to a dataset repository.

In other cases, you might want to use an external storage solution from your Space's code like an external hosted DB, S3, etc.
```

**Review by osanseviero — APPROVED** [review]
Thanks! I left some suggestions

**Comment by radames** [comment]
@XciD I've added more info for Secretes based on your feedback https://github.com/huggingface/hub-docs/pull/575/commits/b4bdafbf2ff7eb7823af1e521a8c37785a447a0f

**Review by christophe-rannou — APPROVED** [review]

File: docs/hub/spaces-sdks-docker.md [hunk 6]
```diff
@@ -17,16 +17,34 @@ app_port: 7860
...
+If you want to expose apps served on multiple ports to the outside world, a workaround is to use a reverse proxy like Nginx to dispatch requests from the broader internet (on a single port) to different internal ports.
 
 ## Secret Management
 
+### Buildtime
+
 In Docker Spaces, the secrets management is different for security reasons. Once you create a secret in the [Settings tab](./spaces-overview#managing-secrets), you can expose the secret by adding the following line in your Dockerfile.
 
+For example, `SECRET_EXAMPLE` is the name of the secret you created in the Settings tab.
+
```
Comment (julien-c) [inline docs/hub/spaces-sdks-docker.md:31]:
```suggestion
For example, if `SECRET_EXAMPLE` is the name of the secret you created in the Settings tab, you can read it at build time by doing `$(cat /run/secrets/SECRET_EXAMPLE)`.

See an example below:

```

what about also adding just an example like

**Review by julien-c — COMMENTED** [review]
left a suggestion to hopefully make Docker secrets clearer

**Review by XciD — APPROVED** [review]

**Pull request merged**